driver-core/src/main/com/mongodb/client/model/changestream/ChangeStreamDocument.java

```diff
@@ -80,23 +81,53 @@ public ChangeStreamDocument(@BsonProperty("resumeToken") final BsonDocument resu
      * @param operationType the operation type
      * @param updateDescription the update description
      */
-    @BsonCreator
+    @Deprecated
     public ChangeStreamDocument(@BsonProperty("resumeToken") final BsonDocument resumeToken,
                                 @BsonProperty("namespace") final MongoNamespace namespace,
                                 @BsonProperty("fullDocument") final TDocument fullDocument,
                                 @BsonProperty("documentKey") final BsonDocument documentKey,
                                 @Nullable @BsonProperty("clusterTime") final BsonTimestamp clusterTime,
                                 @BsonProperty("operationType") final OperationType operationType,
                                 @BsonProperty("updateDescription") final UpdateDescription updateDescription) {
+        this(resumeToken, namespaceToDocument(namespace), fullDocument, documentKey,
+                clusterTime, operationType, updateDescription);
+    }
+
+    /**
+     * Creates a new instance
+     *
+     * @param resumeToken the resume token
+     * @param namespaceDocument the BsonDocument representing the namespace
+     * @param fullDocument the full document
+     * @param documentKey a document containing the _id of the changed document
+     * @param clusterTime the cluster time at which the change occured
+     * @param operationType the operation type
+     * @param updateDescription the update description
+     */
+    @BsonCreator
+    public ChangeStreamDocument(@BsonProperty("resumeToken") final BsonDocument resumeToken,
+                                @BsonProperty("ns") final BsonDocument namespaceDocument,
+                                @BsonProperty("fullDocument") final TDocument fullDocument,
+                                @BsonProperty("documentKey") final BsonDocument documentKey,
+                                @Nullable @BsonProperty("clusterTime") final BsonTimestamp clusterTime,
+                                @BsonProperty("operationType") final OperationType operationType,
+                                @BsonProperty("updateDescription") final UpdateDescription updateDescription) {
         this.resumeToken = resumeToken;
-        this.namespace = namespace;
+        this.namespaceDocument = namespaceDocument;
         this.documentKey = documentKey;
         this.fullDocument = fullDocument;
         this.clusterTime = clusterTime;
         this.operationType = operationType;
         this.updateDescription = updateDescription;
     }
 
+    @BsonIgnore
+    private static BsonDocument namespaceToDocument(final MongoNamespace namespace) {
+        Assertions.notNull("namespace", namespace);
+        return BsonDocument.parse(String.format("{db: '%s', coll: '%s'}",
```

Comment:
Generally we don't use parse to create BsonDocument instances.  Better to construct and then use append method to add fields.  The compiler then makes sure you didn't mess up rather than the run-time (when the json doc is parsed)

Reply:
Done